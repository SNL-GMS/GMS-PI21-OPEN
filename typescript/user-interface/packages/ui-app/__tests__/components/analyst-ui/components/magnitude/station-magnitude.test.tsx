import { CommonTypes, LegacyEventTypes } from '@gms/common-model';
import { render } from '@testing-library/react';
import React from 'react';

import {
  StationMagnitude
  // eslint-disable-next-line max-len
} from '../../../../../src/ts/components/analyst-ui/components/magnitude/components/station-magnitude/station-magnitude-component';
import type {
  StationMagnitudeProps,
  StationMagnitudeSdData,
  StationMagnitudeState
} from '../../../../../src/ts/components/analyst-ui/components/magnitude/components/station-magnitude/types';
import { testMagTypes } from '../../../../__data__/test-util-data';

// eslint-disable-next-line @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports
const Enzyme = require('enzyme');

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

const locationSolution: LegacyEventTypes.LocationSolution = {
  id: '123',
  locationType: 'type',
  locationToStationDistances: [
    {
      azimuth: 1,
      distance: {
        degrees: 1,
        km: 2
      },
      stationId: '1'
    }
  ],
  location: {
    latitudeDegrees: 1,
    longitudeDegrees: 1,
    depthKm: 1,
    time: 1
  },
  featurePredictions: undefined,
  locationRestraint: undefined,
  locationBehaviors: undefined,
  networkMagnitudeSolutions: [
    {
      uncertainty: 1,
      magnitudeType: LegacyEventTypes.MagnitudeType.MB,
      magnitude: 1,
      networkMagnitudeBehaviors: []
    },
    {
      uncertainty: 1,
      magnitudeType: LegacyEventTypes.MagnitudeType.MS,
      magnitude: 1,
      networkMagnitudeBehaviors: []
    },
    {
      uncertainty: 1,
      magnitudeType: LegacyEventTypes.MagnitudeType.MBMLE,
      magnitude: 1,
      networkMagnitudeBehaviors: []
    },
    {
      uncertainty: 1,
      magnitudeType: LegacyEventTypes.MagnitudeType.MSMLE,
      magnitude: 1,
      networkMagnitudeBehaviors: []
    }
  ],
  snapshots: undefined
};

function flushPromises(): any {
  return new Promise(resolve => {
    setTimeout(resolve, 0);
  });
}

describe('when loading station Magnitude', () => {
  const validSignalDetectionForMagnitude = new Map<any, boolean>([
    [LegacyEventTypes.MagnitudeType.MB, true]
  ]);
  const magTypeToAmplitudeMap = new Map<any, StationMagnitudeSdData>([
    [
      LegacyEventTypes.MagnitudeType.MB,
      {
        channel: 'bar',
        phase: CommonTypes.PhaseType.P,
        amplitudeValue: 1,
        amplitudePeriod: 1,
        signalDetectionId: '1',
        time: 1,
        stationName: '1',
        flagForReview: false
      }
    ]
  ]);

  const lazyloadStub: StationMagnitudeProps = {
    checkBoxCallback: jest.fn(),
    setSelectedSdIds: jest.fn(),
    selectedSdIds: ['123'],
    historicalMode: false,
    locationSolution,
    computeNetworkMagnitudeSolution: undefined,
    displayedMagnitudeTypes: testMagTypes,
    amplitudesByStation: [
      {
        stationName: 'foo',
        validSignalDetectionForMagnitude,
        magTypeToAmplitudeMap
      }
    ],
    openEventId: ''
  };
  // eslint-disable-next-line react/jsx-props-no-spreading
  const wrapper = Enzyme.mount(<StationMagnitude {...lazyloadStub} />);

  it.skip('Network station magnitude table renders', () => {
    // eslint-disable-next-line react/jsx-props-no-spreading
    const { container } = render(<StationMagnitude {...lazyloadStub} />);
    expect(container).toMatchSnapshot();
  });

  it.skip('we populate props', () => {
    const props = wrapper.props() as StationMagnitudeProps;
    expect(props).toMatchSnapshot();
  });

  it.skip('we have null initial state', () => {
    const state = wrapper.state() as StationMagnitudeState;
    expect(state).toMatchSnapshot();
  });

  it.skip('we have a onCellClicked function click handler for selecting mb and ms rows', () => {
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'onCellClicked');
    instance.mainTable = {};
    instance.mainTable.getSelectedNodes = jest.fn();
    instance.mainTable.getRenderedNodes = jest.fn(() => [
      { data: { mbSignalDetectionId: '123' }, rowIndex: 1 }
    ]);
    // null call
    instance.onCellClicked(null);
    flushPromises();
    expect(spy).toHaveBeenCalledWith(null);

    // empty call
    instance.onCellClicked({});
    flushPromises();
    expect(spy).toHaveBeenCalledWith({});

    // call with good args
    const defaultClickSpy = jest.spyOn(instance, 'onDefaultClick');
    instance.onCellClicked({
      node: undefined,
      event: {},
      data: { mbSignalDetectionIdevent: '123' },
      column: { colId: 'mbChannel' }
    });
    flushPromises();
    expect(defaultClickSpy).toHaveBeenCalled();
    expect(spy).toHaveBeenCalledWith({
      node: undefined,
      event: {},
      data: { mbSignalDetectionIdevent: '123' },
      column: { colId: 'mbChannel' }
    });

    // called with shiftkey
    const shiftClickSpy = jest.spyOn(instance, 'onShiftClick');
    instance.onCellClicked({
      node: { rowIndex: 1 },
      event: { shiftKey: true },
      data: { mbSignalDetectionIdevent: '123' },
      column: { colId: 'mbChannel' }
    });
    flushPromises();
    expect(shiftClickSpy).toHaveBeenCalled();
    expect(spy).toHaveBeenCalledWith({
      node: { rowIndex: 1 },
      event: { shiftKey: true },
      data: { mbSignalDetectionIdevent: '123' },
      column: { colId: 'mbChannel' }
    });

    // called with ctrlKey/metaKey
    const onControlClickSpy = jest.spyOn(instance, 'onControlClick');
    instance.onCellClicked({
      event: { ctrlKey: true },
      data: { mbSignalDetectionIdevent: '123' },
      column: { colId: 'mbChannel' }
    });
    flushPromises();
    expect(onControlClickSpy).toHaveBeenCalled();
    expect(spy).toHaveBeenCalledWith({
      event: { ctrlKey: true },
      data: { mbSignalDetectionIdevent: '123' },
      column: { colId: 'mbChannel' }
    });
  });

  it.skip('we can getSelectedMbRowIndices', () => {
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'getSelectedMbRowIndices');
    instance.mainTable = {};
    const node = { data: { mbSignalDetectionId: '123' }, rowIndex: 1 };
    instance.mainTable.getRenderedNodes = jest.fn(() => [node]);
    const result = instance.getSelectedMbRowIndices();
    flushPromises();
    expect(result).toEqual([1]);
    expect(spy).toHaveBeenCalled();
  });

  it.skip('we can getSelectedMsRowIndices', () => {
    const instance = wrapper.instance();
    const spy = jest.spyOn(instance, 'getSelectedMsRowIndices');
    instance.mainTable = {};
    const node = { data: { msSignalDetectionId: '123' }, rowIndex: 1 };
    instance.mainTable.getRenderedNodes = jest.fn(() => [node]);
    const result = instance.getSelectedMsRowIndices();
    flushPromises();
    expect(result).toEqual([1]);
    expect(spy).toHaveBeenCalled();
  });
});
