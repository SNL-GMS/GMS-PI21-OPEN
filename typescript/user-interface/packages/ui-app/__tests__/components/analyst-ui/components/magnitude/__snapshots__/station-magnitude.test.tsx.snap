// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading station Magnitude Network station magnitude table renders 1`] = `
<StationMagnitude
  checkBoxCallback={[MockFunction]}
  displayedMagnitudeTypes={Immutable.Map {}}
  historicalMode={false}
  locationSolution={
    Object {
      "featurePredictions": undefined,
      "id": "123",
      "location": Object {
        "depthKm": 1,
        "latitudeDegrees": 1,
        "longitudeDegrees": 1,
        "time": 1,
      },
      "locationBehaviors": undefined,
      "locationRestraint": undefined,
      "locationToStationDistances": Array [
        Object {
          "azimuth": 1,
          "distance": Object {
            "degrees": 1,
            "km": 2,
          },
          "stationId": "1",
        },
      ],
      "locationType": "type",
      "networkMagnitudeSolutions": Array [
        Object {
          "magnitude": 1,
          "magnitudeType": "MB",
          "networkMagnitudeBehaviors": Array [],
          "uncertainty": 1,
        },
        Object {
          "magnitude": 1,
          "magnitudeType": "MS",
          "networkMagnitudeBehaviors": Array [],
          "uncertainty": 1,
        },
        Object {
          "magnitude": 1,
          "magnitudeType": "MBMLE",
          "networkMagnitudeBehaviors": Array [],
          "uncertainty": 1,
        },
        Object {
          "magnitude": 1,
          "magnitudeType": "MSMLE",
          "networkMagnitudeBehaviors": Array [],
          "uncertainty": 1,
        },
      ],
      "snapshots": undefined,
    }
  }
  selectedSdIds={
    Array [
      "123",
    ]
  }
  setSelectedSdIds={[MockFunction]}
  stationsAndSignalDetectionsByStationId={
    Map {
      "1" => Object {
        "distance": 1,
        "lrPhaseDetection": undefined,
        "pPhaseDetection": undefined,
        "station": Object {
          "defaultChannel": Object {
            "depth": 1,
          },
          "id": "1",
          "name": "stringName",
        },
      },
    }
  }
>
  <div
    className="ag-theme-dark table-container"
  >
    <div
      className="list-wrapper"
    >
      <div
        className="max"
      >
        <PureComponent
          columnDefs={
            Array [
              Object {
                "children": Array [
                  Object {
                    "cellStyle": [Function],
                    "field": "station",
                    "filter": true,
                    "headerName": "Station",
                    "pinned": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 70,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "dist",
                    "filter": true,
                    "headerName": "Dist (°)",
                    "pinned": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 80,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "azimuth",
                    "filter": true,
                    "headerComponentFramework": [Function],
                    "headerName": "S-R Azimuth (°)",
                    "pinned": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 110,
                  },
                ],
                "headerName": "Station magnitude",
              },
              Object {
                "children": Array [
                  Object {
                    "cellStyle": [Function],
                    "field": "mbChannel",
                    "filter": true,
                    "headerName": "Channel",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 70,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "mbPhase",
                    "filter": true,
                    "headerName": "Phase",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "mbA5",
                    "filter": true,
                    "headerName": "A5/2",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "mbPeriod",
                    "filter": true,
                    "headerName": "Period (s)",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 80,
                  },
                  Object {
                    "cellRendererFramework": [Function],
                    "cellRendererParams": Object {
                      "magnitudeType": "MB",
                      "padding": 25,
                    },
                    "cellStyle": [Function],
                    "field": "mbDefining",
                    "filter": true,
                    "headerName": "Def",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 50,
                  },
                  Object {
                    "cellStyle": Object {
                      "background-color": "#fff",
                      "text-align": "right",
                    },
                    "field": "mbMag",
                    "filter": true,
                    "headerName": "Mag",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                  Object {
                    "cellStyle": Object {
                      "background-color": "#fff",
                      "text-align": "right",
                    },
                    "field": "mbRes",
                    "filter": true,
                    "headerName": "Res",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                ],
                "headerGroupComponentFramework": [Function],
                "headerGroupComponentParams": Object {
                  "callback": [MockFunction],
                  "definingState": "NONE",
                  "magnitudeType": "MB",
                  "stationIds": Array [],
                },
                "headerName": "Mb",
              },
              Object {
                "children": Array [
                  Object {
                    "cellStyle": [Function],
                    "field": "msChannel",
                    "filter": true,
                    "headerName": "Channel",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 70,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "msPhase",
                    "filter": true,
                    "headerName": "Phase",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "msALR",
                    "filter": true,
                    "headerName": "ALR/2",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "msPeriod",
                    "filter": true,
                    "headerName": "Period (s)",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 80,
                  },
                  Object {
                    "cellRendererFramework": [Function],
                    "cellRendererParams": Object {
                      "magnitudeType": "MS",
                      "padding": 25,
                    },
                    "cellStyle": [Function],
                    "field": "msDefining",
                    "filter": true,
                    "headerName": "Def",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 50,
                  },
                  Object {
                    "cellStyle": Object {
                      "background-color": "#fff",
                      "text-align": "right",
                    },
                    "field": "msMag",
                    "filter": true,
                    "headerName": "Mag",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                  Object {
                    "cellStyle": Object {
                      "background-color": "#fff",
                      "text-align": "right",
                    },
                    "field": "msRes",
                    "filter": true,
                    "headerName": "Res",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                ],
                "headerGroupComponentFramework": [Function],
                "headerGroupComponentParams": Object {
                  "callback": [MockFunction],
                  "definingState": "NONE",
                  "magnitudeType": "MS",
                  "stationIds": Array [],
                },
                "headerName": "Ms",
              },
              Object {
                "children": Array [
                  Object {
                    "cellStyle": [Function],
                    "field": "mbMleChannel",
                    "filter": true,
                    "headerName": "Channel",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 70,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "mbMlePhase",
                    "filter": true,
                    "headerName": "Phase",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "mbMleA5",
                    "filter": true,
                    "headerName": "A5/2",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "mbMlePeriod",
                    "filter": true,
                    "headerName": "Period (s)",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 80,
                  },
                  Object {
                    "cellRendererFramework": [Function],
                    "cellRendererParams": Object {
                      "magnitudeType": "MBMLE",
                      "padding": 25,
                    },
                    "cellStyle": [Function],
                    "field": "mbMleDefining",
                    "filter": true,
                    "headerName": "Def",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 50,
                  },
                  Object {
                    "cellStyle": Object {
                      "background-color": "#fff",
                      "text-align": "right",
                    },
                    "field": "mbMleMag",
                    "filter": true,
                    "headerName": "Mag",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                  Object {
                    "cellStyle": Object {
                      "background-color": "#fff",
                      "text-align": "right",
                    },
                    "field": "mbMleRes",
                    "filter": true,
                    "headerName": "Res",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                ],
                "headerGroupComponentFramework": [Function],
                "headerGroupComponentParams": Object {
                  "callback": [MockFunction],
                  "definingState": "NONE",
                  "magnitudeType": "MBMLE",
                  "stationIds": Array [],
                },
                "headerName": "Mb Mle",
              },
              Object {
                "children": Array [
                  Object {
                    "cellStyle": [Function],
                    "field": "msMleChannel",
                    "filter": true,
                    "headerName": "Channel",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 70,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "msMlePhase",
                    "filter": true,
                    "headerName": "Phase",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "msMleALR",
                    "filter": true,
                    "headerName": "ALR/2",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "msMlePeriod",
                    "filter": true,
                    "headerName": "Period (s)",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 80,
                  },
                  Object {
                    "cellRendererFramework": [Function],
                    "cellRendererParams": Object {
                      "magnitudeType": "MSMLE",
                      "padding": 25,
                    },
                    "cellStyle": [Function],
                    "field": "msMleDefining",
                    "filter": true,
                    "headerName": "Def",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 50,
                  },
                  Object {
                    "cellStyle": Object {
                      "background-color": "#fff",
                      "text-align": "right",
                    },
                    "field": "msMleMag",
                    "filter": true,
                    "headerName": "Mag",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                  Object {
                    "cellStyle": [Function],
                    "field": "msMleRes",
                    "filter": true,
                    "headerName": "Res",
                    "hide": true,
                    "resizable": false,
                    "sortable": true,
                    "width": 60,
                  },
                ],
                "headerGroupComponentFramework": [Function],
                "headerGroupComponentParams": Object {
                  "callback": [MockFunction],
                  "definingState": "NONE",
                  "magnitudeType": "MSMLE",
                  "stationIds": Array [],
                },
                "headerName": "Ms Mle",
              },
            ]
          }
          context={Object {}}
          immutableData={true}
          getRowNodeId={[Function]}
          onCellClicked={[Function]}
          onGridReady={[Function]}
          overlayNoRowsTemplate="No associated signal detections with appropriate phases"
          rowData={
            Array [
              Object {
                "azimuth": "1.000",
                "checkBoxCallback": [MockFunction],
                "dist": "1.000",
                "historicalMode": false,
                "id": "1",
                "mbA5": "",
                "mbChannel": "",
                "mbDefining": false,
                "mbMag": undefined,
                "mbMleA5": "",
                "mbMleChannel": "",
                "mbMleDefining": false,
                "mbMleMag": undefined,
                "mbMlePeriod": "",
                "mbMlePhase": "",
                "mbMleRes": undefined,
                "mbPeriod": "",
                "mbPhase": "",
                "mbRes": undefined,
                "mbSignalDetectionId": "",
                "msALR": "",
                "msChannel": "",
                "msDefining": false,
                "msMag": undefined,
                "msMleALR": "",
                "msMleChannel": "",
                "msMleDefining": false,
                "msMleMag": undefined,
                "msMlePeriod": "",
                "msMlePhase": "",
                "msMleRes": undefined,
                "msPeriod": "",
                "msPhase": "",
                "msRes": undefined,
                "msSignalDetectionId": "",
                "selectedSdIds": Array [
                  "123",
                ],
                "station": "stringName",
                "stationId": "1",
              },
            ]
          }
          rowDeselection={true}
          rowSelection="none"
          suppressContextMenu={true}
        >
          <AgGridReact
            columnDefs={
              Array [
                Object {
                  "children": Array [
                    Object {
                      "cellStyle": [Function],
                      "field": "station",
                      "filter": true,
                      "headerName": "Station",
                      "pinned": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 70,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "dist",
                      "filter": true,
                      "headerName": "Dist (°)",
                      "pinned": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 80,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "azimuth",
                      "filter": true,
                      "headerComponentFramework": [Function],
                      "headerName": "S-R Azimuth (°)",
                      "pinned": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 110,
                    },
                  ],
                  "headerName": "Station magnitude",
                },
                Object {
                  "children": Array [
                    Object {
                      "cellStyle": [Function],
                      "field": "mbChannel",
                      "filter": true,
                      "headerName": "Channel",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 70,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "mbPhase",
                      "filter": true,
                      "headerName": "Phase",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "mbA5",
                      "filter": true,
                      "headerName": "A5/2",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "mbPeriod",
                      "filter": true,
                      "headerName": "Period (s)",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 80,
                    },
                    Object {
                      "cellRendererFramework": [Function],
                      "cellRendererParams": Object {
                        "magnitudeType": "MB",
                        "padding": 25,
                      },
                      "cellStyle": [Function],
                      "field": "mbDefining",
                      "filter": true,
                      "headerName": "Def",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 50,
                    },
                    Object {
                      "cellStyle": Object {
                        "background-color": "#fff",
                        "text-align": "right",
                      },
                      "field": "mbMag",
                      "filter": true,
                      "headerName": "Mag",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                    Object {
                      "cellStyle": Object {
                        "background-color": "#fff",
                        "text-align": "right",
                      },
                      "field": "mbRes",
                      "filter": true,
                      "headerName": "Res",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                  ],
                  "headerGroupComponentFramework": [Function],
                  "headerGroupComponentParams": Object {
                    "callback": [MockFunction],
                    "definingState": "NONE",
                    "magnitudeType": "MB",
                    "stationIds": Array [],
                  },
                  "headerName": "Mb",
                },
                Object {
                  "children": Array [
                    Object {
                      "cellStyle": [Function],
                      "field": "msChannel",
                      "filter": true,
                      "headerName": "Channel",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 70,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "msPhase",
                      "filter": true,
                      "headerName": "Phase",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "msALR",
                      "filter": true,
                      "headerName": "ALR/2",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "msPeriod",
                      "filter": true,
                      "headerName": "Period (s)",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 80,
                    },
                    Object {
                      "cellRendererFramework": [Function],
                      "cellRendererParams": Object {
                        "magnitudeType": "MS",
                        "padding": 25,
                      },
                      "cellStyle": [Function],
                      "field": "msDefining",
                      "filter": true,
                      "headerName": "Def",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 50,
                    },
                    Object {
                      "cellStyle": Object {
                        "background-color": "#fff",
                        "text-align": "right",
                      },
                      "field": "msMag",
                      "filter": true,
                      "headerName": "Mag",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                    Object {
                      "cellStyle": Object {
                        "background-color": "#fff",
                        "text-align": "right",
                      },
                      "field": "msRes",
                      "filter": true,
                      "headerName": "Res",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                  ],
                  "headerGroupComponentFramework": [Function],
                  "headerGroupComponentParams": Object {
                    "callback": [MockFunction],
                    "definingState": "NONE",
                    "magnitudeType": "MS",
                    "stationIds": Array [],
                  },
                  "headerName": "Ms",
                },
                Object {
                  "children": Array [
                    Object {
                      "cellStyle": [Function],
                      "field": "mbMleChannel",
                      "filter": true,
                      "headerName": "Channel",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 70,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "mbMlePhase",
                      "filter": true,
                      "headerName": "Phase",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "mbMleA5",
                      "filter": true,
                      "headerName": "A5/2",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "mbMlePeriod",
                      "filter": true,
                      "headerName": "Period (s)",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 80,
                    },
                    Object {
                      "cellRendererFramework": [Function],
                      "cellRendererParams": Object {
                        "magnitudeType": "MBMLE",
                        "padding": 25,
                      },
                      "cellStyle": [Function],
                      "field": "mbMleDefining",
                      "filter": true,
                      "headerName": "Def",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 50,
                    },
                    Object {
                      "cellStyle": Object {
                        "background-color": "#fff",
                        "text-align": "right",
                      },
                      "field": "mbMleMag",
                      "filter": true,
                      "headerName": "Mag",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                    Object {
                      "cellStyle": Object {
                        "background-color": "#fff",
                        "text-align": "right",
                      },
                      "field": "mbMleRes",
                      "filter": true,
                      "headerName": "Res",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                  ],
                  "headerGroupComponentFramework": [Function],
                  "headerGroupComponentParams": Object {
                    "callback": [MockFunction],
                    "definingState": "NONE",
                    "magnitudeType": "MBMLE",
                    "stationIds": Array [],
                  },
                  "headerName": "Mb Mle",
                },
                Object {
                  "children": Array [
                    Object {
                      "cellStyle": [Function],
                      "field": "msMleChannel",
                      "filter": true,
                      "headerName": "Channel",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 70,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "msMlePhase",
                      "filter": true,
                      "headerName": "Phase",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "msMleALR",
                      "filter": true,
                      "headerName": "ALR/2",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "msMlePeriod",
                      "filter": true,
                      "headerName": "Period (s)",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 80,
                    },
                    Object {
                      "cellRendererFramework": [Function],
                      "cellRendererParams": Object {
                        "magnitudeType": "MSMLE",
                        "padding": 25,
                      },
                      "cellStyle": [Function],
                      "field": "msMleDefining",
                      "filter": true,
                      "headerName": "Def",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 50,
                    },
                    Object {
                      "cellStyle": Object {
                        "background-color": "#fff",
                        "text-align": "right",
                      },
                      "field": "msMleMag",
                      "filter": true,
                      "headerName": "Mag",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                    Object {
                      "cellStyle": [Function],
                      "field": "msMleRes",
                      "filter": true,
                      "headerName": "Res",
                      "hide": true,
                      "resizable": false,
                      "sortable": true,
                      "width": 60,
                    },
                  ],
                  "headerGroupComponentFramework": [Function],
                  "headerGroupComponentParams": Object {
                    "callback": [MockFunction],
                    "definingState": "NONE",
                    "magnitudeType": "MSMLE",
                    "stationIds": Array [],
                  },
                  "headerName": "Ms Mle",
                },
              ]
            }
            context={Object {}}
            immutableData={true}
            getRowNodeId={[Function]}
            onCellClicked={[Function]}
            onGridReady={[Function]}
            overlayNoRowsTemplate="No associated signal detections with appropriate phases"
            rowData={
              Array [
                Object {
                  "azimuth": "1.000",
                  "checkBoxCallback": [MockFunction],
                  "dist": "1.000",
                  "historicalMode": false,
                  "id": "1",
                  "mbA5": "",
                  "mbChannel": "",
                  "mbDefining": false,
                  "mbMag": undefined,
                  "mbMleA5": "",
                  "mbMleChannel": "",
                  "mbMleDefining": false,
                  "mbMleMag": undefined,
                  "mbMlePeriod": "",
                  "mbMlePhase": "",
                  "mbMleRes": undefined,
                  "mbPeriod": "",
                  "mbPhase": "",
                  "mbRes": undefined,
                  "mbSignalDetectionId": "",
                  "msALR": "",
                  "msChannel": "",
                  "msDefining": false,
                  "msMag": undefined,
                  "msMleALR": "",
                  "msMleChannel": "",
                  "msMleDefining": false,
                  "msMleMag": undefined,
                  "msMlePeriod": "",
                  "msMlePhase": "",
                  "msMleRes": undefined,
                  "msPeriod": "",
                  "msPhase": "",
                  "msRes": undefined,
                  "msSignalDetectionId": "",
                  "selectedSdIds": Array [
                    "123",
                  ],
                  "station": "stringName",
                  "stationId": "1",
                },
              ]
            }
            rowDeselection={true}
            rowSelection="none"
            suppressAsyncEvents={true}
            suppressContextMenu={true}
          >
            <div
              style={
                Object {
                  "height": "100%",
                }
              }
            />
          </AgGridReact>
        </PureComponent>
      </div>
    </div>
  </div>
</StationMagnitude>
`;

exports[`when loading station Magnitude we have null initial state 1`] = `null`;

exports[`when loading station Magnitude we populate props 1`] = `
Object {
  "checkBoxCallback": [MockFunction],
  "computeNetworkMagnitudeSolution": undefined,
  "displayedMagnitudeTypes": Immutable.Map {},
  "historicalMode": false,
  "locationSolution": Object {
    "featurePredictions": undefined,
    "id": "123",
    "location": Object {
      "depthKm": 1,
      "latitudeDegrees": 1,
      "longitudeDegrees": 1,
      "time": 1,
    },
    "locationBehaviors": undefined,
    "locationRestraint": undefined,
    "locationToStationDistances": Array [
      Object {
        "azimuth": 1,
        "distance": Object {
          "degrees": 1,
          "km": 2,
        },
        "stationId": "1",
      },
    ],
    "locationType": "type",
    "networkMagnitudeSolutions": Array [
      Object {
        "magnitude": 1,
        "magnitudeType": "MB",
        "networkMagnitudeBehaviors": Array [],
        "uncertainty": 1,
      },
      Object {
        "magnitude": 1,
        "magnitudeType": "MS",
        "networkMagnitudeBehaviors": Array [],
        "uncertainty": 1,
      },
      Object {
        "magnitude": 1,
        "magnitudeType": "MBMLE",
        "networkMagnitudeBehaviors": Array [],
        "uncertainty": 1,
      },
      Object {
        "magnitude": 1,
        "magnitudeType": "MSMLE",
        "networkMagnitudeBehaviors": Array [],
        "uncertainty": 1,
      },
    ],
    "snapshots": undefined,
  },
  "selectedSdIds": Array [
    "123",
  ],
  "setSelectedSdIds": [MockFunction],
  "stationsAndSignalDetectionsByStationId": Map {
    "1" => Object {
      "distance": 1,
      "lrPhaseDetection": undefined,
      "pPhaseDetection": undefined,
      "station": Object {
        "defaultChannel": Object {
          "depth": 1,
        },
        "id": "1",
        "name": "stringName",
      },
    },
  },
}
`;
